### PR TITLE
Improve dba::selectFirst calls

### PR DESCRIPTION
**Pull request by MrPetovan**
Part of #4176

While doing #4205, I realized that some calls to `dba::selectFirst` weren't cleanly done. I was able to find some remaining references to the previous `$r[0]` record.

I decided to go on a mgical journey, and visit all current `dba::selectFirst()` calls and either improve the legibility by renaming the return variable or fixing an eventual forgotten reference.

### CODE REVIEW

**Comment by annando** [comment]
Is it some urgent bugfix or has it time to be reviewed this evening (European time)?

**Comment by MrPetovan** [comment]
It has some bug fixes but nobody complained so far, so we're good on the time frame.

File: include/identity.php [hunk 1]
```diff
@@ -185,11 +184,11 @@ function get_profiledata_by_nick($nickname, $uid = 0, $profile = 0)
 			INNER JOIN `user` ON `profile`.`uid` = `user`.`uid`
 			WHERE `user`.`nickname` = ? AND `profile`.`id` = ? LIMIT 1",
 			$nickname,
-			$profile_int
+			intval($profile_id)
```
Comment (annando) [inline include/identity.php]: This ```intval``` is superfluous.

File: include/message.php [hunk 2]
```diff
@@ -67,11 +67,8 @@ function send_message($recipient = 0, $body = '', $subject = '', $replyto = '')
 		$fields = array('uid' => local_user(), 'guid' => $conv_guid, 'creator' => $sender_handle,
 			'created' => datetime_convert(), 'updated' => datetime_convert(),
 			'subject' => $subject, 'recips' => $handles);
-		dba::insert('conv', $fields);
-
-		$r = dba::selectFirst('conv', ['id'], ['guid' => $conv_guid, 'uid' => local_user()]);
-		if (DBM::is_result($r)) {
-			$convid = $r['id'];
+		if (dba::insert('conv', $fields)) {
+			$convid = dba::lastInsertId();
```
Comment (annando) [inline include/message.php]: That's much too easy using that function ;-)

File: mod/network.php [hunk 3]
```diff
@@ -596,12 +596,12 @@ function networkThreadedView(App $a, $update = 0) {
 			'allow_location' => $a->user['allow_location'],
 			'default_location' => $a->user['default-location'],
 			'nickname' => $a->user['nickname'],
-			'lockstate'=> ((($group) || ($cid) || ($nets) || (is_array($a->user) &&
+			'lockstate'=> ((($gid) || ($cid) || ($nets) || (is_array($a->user) &&
```
Comment (annando) [inline mod/network.php]: You could have removed some of the brackets :-)

File: include/message.php (hunk 2 above)
Comment (MrPetovan) [reply include/message.php]: I know right! Programming should only about pain and suffering!

**Pull request merged**